aptos-move/framework/supra-framework/sources/poel/poel.move

```diff
@@ -0,0 +1,308 @@
+        // Index of the last withdrawal request submitted by the admin
+        withdraw_OLC_index: u64
+    }
+ 
+    /// This struct holds the field items (total_borrow_requests & total_withdraw_requests) mapped to requests
+    /// table (AssetId => BorrowWithdrawRequestTableItems) in the BorrowWithdrawRequest
+    struct BorrowWithdrawRequestTableItems {
+        total_borrow_requests: u64,
+        total_withdraw_requests: u64,
+    }
+
+
+    /// Maps each AssetID to a tuple containing the total amounts of withdraw requests submitted to the PoEL contract.
+    /// Borrow requests are initiated by users to borrow Supra and mint iAssets after depositing their assets in intralayer vaults.
+    /// Withdraw requests are made by users aiming to retrieve their assets from intralayer vaults in the subsequent cycle.
+    /// Both types of requests are processed in the next cycle.
+    struct BorrowWithdrawRequest {
+        // Maps each AssetID to a BorrowWithdrawRequestTableItems
+        requests: table::Table<u64, BorrowWithdrawRequestTableItems>,
+    }
+
+    //Functions:
+
+    #[view]
+    public fun get_total_liquidity_current_cycle_index(): u64 {
+        0
+    }
+
+    #[view]
+    public fun get_total_liquidity(): address {
+        @supra_framework
+    }
+
+
+    /// adds an asset with its coresponding information to the System
+    public fun add_asset(
+        _total_liquidity: &mut TotalLiquidity,
+        _asset_id: u64,
+        _asset_name: vector<u8>,
+        _asset_supply: u64,
+        _iAsset_supply: u64,
+        _collateralisation_rate: u64,
+        _desired_weights: u64,
+        _desirability_score: u64,
+        _asset_price: u64
+    ) {
+    }
+
+    /// Adds a staking pool to the existing pools for delegation os $supra
+    public fun add_staking_pool(
+        _delegated_amount: &mut DelegatedAmount,
+        _pool_id: u64,
+        _pool_address: address,
+        _delegated_amount_value: u64,
+        _pending_inactive_balance: u64
+    ) {
+    }
+
+
+    public fun submit_borrow_request(
+        _request_table: &mut BorrowWithdrawRequest,
+        _asset_id: u64,
+        _borrow_amount: u64
+    ) {
+
+    }
+
+
+    public fun submit_withdraw_request(
+        _request_table: &mut BorrowWithdrawRequest,
+        _asset_id: u64,
+        _withdraw_amount: u64
+    ) {
+
+    }
+
+    /// should return a &BorrowRequestTableItems
+    public fun get_requests(
+        _request_table: &BorrowWithdrawRequest,
+        _asset_id: u64
+    ) {
+    }
+
+
+    // Updates asset weights in the TotalLiquidity struct to align with current strategic objectives.
+    public fun update_desired_weight(
+        _collaterisation_rate_vector: vector<u64>,
+        _account: &signer
+    ) {
+
+    }
+
+    /// Updates desirability scores for all assets in the TotalLiquidity struct.
+    public fun batch_update_desirability_score(
+        _desirability_score_vector: vector<u64>,
+        _account: &signer
```

Comment:
Improve function header comments, write role of each parameter, `signer` should come first